db/db_impl/db_impl_write.cc

```diff
@@ -1933,11 +2113,46 @@ Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context) {
   }
 
   cfd->mem()->SetNextLogNumber(logfile_number_);
-  cfd->imm()->Add(cfd->mem(), &context->memtables_to_free_);
+  // If MemPurge activated, purge and delete current memtable.
+  if (immutable_db_options_.experimental_allow_mempurge &&
+      (new_mem != nullptr) &&
+      ((cfd->GetFlushReason() == FlushReason::kOthers) ||
+       (cfd->GetFlushReason() == FlushReason::kManualFlush))) {
+    Status mempurge_s = MemPurge(cfd, new_mem);
```

Comment:
This is walking the memtable while holding the DB mutex. In production this would lead to many DB operations blocking for potentially a second or more during purge. This is why it needs to be a variant of (background) flush. At least eventually

Reply:
Offline discussion: This limitation will be noted in the PR summary, indicating the intended direction of future work.